### PR TITLE
wr-72: Create genres table

### PR DESCRIPTION
**Pull request by IrynaK-a**

### CODE REVIEW

File: backend/src/db/migrations/20230828103237_create_genres_table.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { type Knex } from 'knex';
+
+import { GENRES } from './libs/constants/genres.js';
```
Comment (andanf-e) [inline backend/src/db/migrations/20230828103237_create_genres_table.ts]: Place them directly in migration file

File: backend/src/db/migrations/libs/constants/genres.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+const GENRES = [
+  { name: 'Science Fiction', key: 'sci-fi' },
```
Comment (andanf-e) [inline backend/src/db/migrations/libs/constants/genres.ts]: Use snake_case for keys, change it for all keys with kebab-case
```suggestion
  { name: 'Science Fiction', key: 'sci_fi' },
```

**Review by andanf-e — CHANGES_REQUESTED** [review]

**Review by andanf-e — DISMISSED** [review]

File: backend/src/db/migrations/20230828103237_create_genres_table.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  { name: 'Non-fiction', key: 'non_fiction' },
+  { name: 'Paranormal', key: 'paranormal' },
+  { name: 'Satire', key: 'satire' },
+  { name: 'Young Adult', key: 'ya_fiction' },
+  { name: 'Crime', key: 'crime' },
+  { name: 'Drama', key: 'drama' },
+  { name: 'Science', key: 'science' },
+  { name: 'Urban Fantasy', key: 'urban_fantasy' },
+  { name: 'Memoir', key: 'memoir' },
+];
```
Comment (NazarKovtun1) [inline backend/src/db/migrations/20230828103237_create_genres_table.ts:24]: Should not it be moved to seeds?

File: backend/src/packages/genres/genre.entity.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+  private constructor({
+    id,
+    name,
+    key,
+  }: {
+    id: number;
+    name: string;
+    key: string;
+  }) {
```
Comment (NazarKovtun1) [inline backend/src/packages/genres/genre.entity.ts]: Please, create a type for it and put it right in this file (right before GenreEntity)

Comment (NazarKovtun1) [reply backend/src/packages/genres/genre.entity.ts]: Also, check all other places, where this case with type is not covered (not just method parameters but return types as well)

**Review by NazarKovtun1 — CHANGES_REQUESTED** [review]

File: backend/src/db/migrations/20230828103237_create_genres_table.ts (hunk 3 above)
Comment (andanf-e) [reply backend/src/db/migrations/20230828103237_create_genres_table.ts:24]: Nope, seeds are not the best choice for deployed app data, as it should be consistent

File: backend/src/packages/genres/genre.entity.ts (hunk 4 above)
Comment (IrynaK-a) [reply backend/src/packages/genres/genre.entity.ts]: Do you mean in other entities too?

**Review by andanf-e — DISMISSED** [review]

File: backend/src/packages/genres/genre.entity.ts (hunk 4 above)
Comment (mhrechkin) [reply backend/src/packages/genres/genre.entity.ts]: @WilhelmWesser i think we use the same approach in all entities, so there is no need to change it in this pr

Comment (NazarKovtun1) [reply backend/src/packages/genres/genre.entity.ts]: Okey then

**Review by NazarKovtun1 — DISMISSED** [review]

**Pull request merged**